File: 800/2183A/main.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
    int t;
    cin>>t;
    
    while(t--)
    {
        int n;
        cin>>n;
        
        int a;
        cin>>a;
        
        for(int i=1;i<n-1;i++)
        {
            int b;
            cin>>b;
        }
        
        int b;
        cin>>b;
        
        if(a == b && a == 0)
            cout<<"BOB"<<endl;
        else
            cout<<"ALICE"<<endl;
    }
    
    return 0;
}

File: 800/2183A/main.java
import java.util.*;
public class main 
{
    public static void main(String[] args) 
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();  // number of test cases

        while (t-- > 0) 
        {
            int n = sc.nextInt();

            int a = sc.nextInt();  // first element

            // read middle elements (ignored in logic)
            for (int i = 1; i < n - 1; i++)
            {
                int temp = sc.nextInt();
            }

            int b = sc.nextInt();  // last element

            if (a == b && a == 0)
                System.out.println("BOB");
            else
                System.out.println("ALICE");
        }
    }
}
